### PR TITLE
Define mongoose schemas for Users, Threads and Messages

### PR DESCRIPTION
**Pull request by ajzawawi**
For review please. Completed second item in task list. #2

### CODE REVIEW

File: api/db/thread.js [hunk 1]
```diff
@@ -0,0 +1,12 @@
+import mongoose from 'mongoose';
+
+const threadSchema = new mongoose.Schema({
+  name: String,
+  isRead: Boolean,
```
Comment (RamyElkest) [inline api/db/thread.js]: can you replace with with readBy: \[\] (I found out scalar isRead on a thread doesn't actually make much sense for a 2+ party room)

File: api/db/thread.js [hunk 2]
```diff
@@ -0,0 +1,12 @@
...
+  messages: []
```
Comment (RamyElkest) [inline api/db/thread.js]: I'm not too sure which design is preferable (nested vs flat+ref). But from experience I tend to think it's better to decouple them. aka just have a threadId on the message instead

I'm happy to discuss if you disagree :+1:

File: api/db/thread.js [hunk 3]
```diff
@@ -0,0 +1,12 @@
...
+}, {
+  timestamps: true
+});
+
+
+export default mongoose.model('User', userSchema);
```
Comment (RamyElkest) [inline api/db/thread.js]: Thread

**Comment by ajzawawi** [comment]
@RamyElkest Check out the updated. It makes sense to decouple them.

File: api/db/thread.js [hunk 4]
```diff
@@ -2,11 +2,11 @@ import mongoose from 'mongoose';
 
 const threadSchema = new mongoose.Schema({
   name: String,
-  isRead: Boolean,
-  messages: []
+  readBy: [],
+  threadId: String
```
Comment (RamyElkest) [inline api/db/thread.js]: Thanks for the reactiveness!

threadId will go on the message

**Comment by RamyElkest** [comment]
also running ```node_modules/eslint/bin/eslint.js --fix api``` should fix the build

**Comment by ajzawawi** [comment]
@RamyElkest Build is passing now.

**Comment by RamyElkest** [comment]
:100: thanks @ajzawawi

**Pull request merged**